simulation/traffic_simulator/src/behavior/longitudinal_speed_planning.cpp

```diff
@@ -0,0 +1,326 @@
+    if (std::abs(v - target_speed) >= 0.01) {
+      return duration;
+    }
+    // While quadratic acceleration, the entity reached the target speed.
+    else {
+      return getQuadraticAccelerationDurationWithConstantJerk(
+        target_speed, constraints, current_twist, current_accel);
+    }
+  } else {
+    double duration =
+      (current_accel.linear.x - constraints.max_deceleration) / constraints.max_deceleration_rate;
+    double v = getVelocityWithConstantJerk(
+      current_twist, current_accel, -constraints.max_deceleration_rate, duration);
+    // While quadratic acceleration, the entity does not reached the target speed.
+    if (std::abs(v - target_speed) >= 0.01) {
+      return duration;
+    }
+    // While quadratic acceleration, the entity reached the target speed.
+    else {
+      return getQuadraticAccelerationDurationWithConstantJerk(
+        target_speed, constraints, current_twist, current_accel);
+    }
+  }
+}
+
+auto LongitudinalSpeedPlanner::getLinearAccelerationDuration(
+  double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints & constraints,
+  const geometry_msgs::msg::Twist & current_twist,
+  const geometry_msgs::msg::Accel & current_accel) const -> double
+{
+  const double quad_duration =
+    getQuadraticAccelerationDuration(target_speed, constraints, current_twist, current_accel);
+  if (isReachedToTargetSpeedWithConstantJerk(
+        target_speed, constraints, current_twist, current_accel, quad_duration)) {
+    return 0;
+  } else {
+    geometry_msgs::msg::Twist twist = current_twist;
+    twist.linear.x = getVelocityWithConstantJerk(
+      target_speed, current_twist, current_accel, constraints, quad_duration);
+    if (isAccelerating(target_speed, current_twist)) {
+      return (target_speed - twist.linear.x) / constraints.max_acceleration;
+    } else {
+      return (twist.linear.x - target_speed) / constraints.max_deceleration;
+    }
+  }
+}
+
+auto LongitudinalSpeedPlanner::isReachedToTargetSpeedWithConstantJerk(
+  double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints & constraints,
+  const geometry_msgs::msg::Twist & current_twist, const geometry_msgs::msg::Accel & current_accel,
+  double duration, double tolerance) const -> bool
+{
+  if (
+    std::abs(
+      getVelocityWithConstantJerk(
+        target_speed, current_twist, current_accel, constraints, duration) -
+      target_speed) <= std::abs(tolerance)) {
+    return true;
+  }
+  return false;
+}
+
+auto LongitudinalSpeedPlanner::isAccelerating(
+  double target_speed, const geometry_msgs::msg::Twist & current_twist) const -> bool
+{
+  if (current_twist.linear.x > target_speed) {
+    return false;
+  }
+  return true;
+}
+
+auto LongitudinalSpeedPlanner::isDecelerating(
+  double target_speed, const geometry_msgs::msg::Twist & current_twist) const -> bool
+{
+  if (current_twist.linear.x < target_speed) {
+    return false;
+  }
+  return true;
```

Comment:
Consider simply rewriting it like this:
```
return not (current_twist.linear.x < target_speed);
```

Reply:
update to `return (current_twist.linear.x <= target_speed);`

Reply:
The updated code has the opposite semantics, is this intentional?
I think its a mistake.